commons/src/scanner/mod.rs

```diff
@@ -56,7 +34,18 @@ where
     }
 }
 
-impl<P> Scanner<P> {
+impl Scanner {
+    pub fn try_new() -> Result<Self, SegmenterError> {
+        let icu_data_provider = IcuDataProvider;
+        let fallback_provider = LocaleFallbackProvider::try_new_unstable(icu_data_provider)?;
+        let segmenter = GraphemeClusterSegmenter::try_new_unstable(&fallback_provider)?;
```

Comment:
Is it known under which circumstances does this fail? Maybe we can change this function to be infallible (return `Scanner` directly without wrapping it into `Result`) now that we use module as data and can ensure compatibility.

Reply:
I am not sure. 
it definitely fails if the generated module and used crate versions are not in-sync, but there could also be other reasons. would there be much benefit to make scanner infallible?
I assume making it infallible means panicking if it fails, but in our case if creating the scammer fails, we kind of also stop everything 🤔 so there is not that much of a difference?

Reply:
If that's the only reason it could fail, then I would say it would almost be better to simply panic at this point. That is one of the cases where we can't recover from the error in any meaningful way, so crashing it is better option to catch the problem as soon as possible. 

It would be best if we can fail at compile time, so that failure is really ruled out if we ship a compiled binary. But I'm not sure if that's possible at all.

Reply:
maybe using once_cell to create a static scanner? 😅

Reply:
Alternative to panic would be to log errors inside of this function and then call [`std::process::exit(1)`](https://doc.rust-lang.org/std/process/fn.exit.html) or something like that.

Reply:
> maybe using once_cell to create a static scanner? 😅

I don't think that would help. We would need the actual provider and segmenter to be constructed at compile time, but their functions are not `const`, so that wont work.

Reply:
I think that would be a nice solution, with error logging. We could also include some debug logs to provide more information on what's gone wrong. 

However, I suggest we make an issue for that one and investigate a little more. It might be the case that these functions return error in more circumstances than what we assume right now. This is fine for now 👍🏻